python/ray/experimental/collective/conftest.py

```diff
@@ -0,0 +1,250 @@
+import copy
+import uuid
+from typing import Dict, FrozenSet, List, Optional, Set, Tuple
+
+import torch
+
+import ray
+from ray.experimental.channel.common import ChannelContext
+from ray.experimental.channel.gpu_communicator import (
+    GPUCommunicator,
+    ReduceOp,
+    TorchTensorAllocator,
+)
+
+
+class AbstractNcclGroup(GPUCommunicator):
```

Comment:
Need to move this from test_collective_dag.py, otherwise there is the following error:

```
>                       raise value.as_instanceof_cause()
E                       ray.exceptions.RayTaskError(RaySystemError): ray::CPUTorchTensorWorker.__ray_call__() (pid=2553478, ip=172.31.15.128, actor_id=a378363d53bded77aa33e28301000000, repr=<test_collective_dag.CPUTorchTensorWorker object at 0x7f02403f7c40>)
E                         At least one of the input arguments for this task could not be computed:
E                       ray.exceptions.RaySystemError: System error: No module named 'test_collective_dag'
E                       traceback: Traceback (most recent call last):
E                       ModuleNotFoundError: No module named 'test_collective_dag'
```